LanguageFeatures/Class-modifiers/syntax_A06_t02.dart

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Many combinations don't make sense:
+/// ...
+/// - mixin as a modifier can obviously only be applied to a class declaration,
+///   which makes it also introduce a mixin declaration.
+///
+/// @description Check that it is a compile-time error if `mixin class` extends
+/// class other than Object
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=class-modifiers
+
+class C {}
+
+mixin class MixinClass extends C {
```

Comment:
Even `mixin class MixinClass extends Object {}` should probably be an error (it may or may not be a syntax error), because there is very little value in supporting that particular construct.

Reply:
Ups... Spec:

> It's a compile-time error if a mixin class declaration:
> ...
> - has an extends clause,

It's an issue that `mixin class MixinClassExtendsObject extends Object {}` is not an error

Reply:
Good catch! So let's test that case as well.